_inc/connect-button.js

```diff
@@ -77,7 +77,7 @@ jQuery( document ).ready( function ( $ ) {
 					_wpnonce: jpConnect.apiNonce,
 				},
 				error: jetpackConnectButton.handleConnectionError,
-				success: jetpackConnectButton.handleConnectionSuccess,
+				success: jetpackConnectButton.handleAuthorizationComplete,
```

Comment:
This is the hacky magic that straight up skips the auth in iframe, and redirect right to plans grid.